src/sidebar/services/annotations.js

```diff
@@ -59,7 +61,66 @@ export default function annotations(
     searchClient.get({ uri: uris, group: groupId });
   }
 
+  /**
+   * Apply changes for the given `annotation` from its draft in the store (if
+   * any) and return a new object with those changes integrated.
+   */
+  function applyDraftChanges(annotation) {
+    const changes = {};
+    const draft = store.getDraft(annotation);
+
+    if (draft) {
+      changes.tags = draft.tags;
+      changes.text = draft.text;
+      changes.permissions = draft.isPrivate
+        ? privatePermissions(annotation.user)
+        : sharedPermissions(annotation.user, annotation.group);
+    }
+
+    // Integrate changes from draft into object to be persisted
+    return Object.assign({}, annotation, changes);
+  }
+
+  /**
+   * Save new (or update existing) annotation. On success,
+   * the annotation's `Draft` will be removed and the annotation added
+   * to the store.
+   */
+  function save(annotation) {
+    let saved;
+
+    const annotationWithChanges = applyDraftChanges(annotation);
+
+    if (isNew(annotation)) {
+      saved = api.annotation.create({}, annotationWithChanges);
+    } else {
+      saved = api.annotation.update(
+        { id: annotation.id },
+        annotationWithChanges
+      );
+    }
+
+    return saved.then(savedAnnotation => {
+      // Copy across internal properties which are not part of the annotation
+      // model saved on the server
+      savedAnnotation.$tag = annotation.$tag;
+      Object.keys(annotation).forEach(function(k) {
+        if (k[0] === '$') {
+          savedAnnotation[k] = annotation[k];
+        }
+      });
+
+      // Clear out any pending changes (draft)
+      store.removeDraft(annotation);
+
+      // Add (or, in effect, update) the annotation to the store's collection
+      store.addAnnotations([].concat(savedAnnotation));
```

Comment:
Any reason not to simplify `[].concat(savedAnnotation)` to just `[savedAnnotation]`?

Reply:
Hah! I forgot to ask the same question myself in my own comments. I spotted this and was like, hum, I wonder if there was some particular reason this was done. Yeah, I'll change it.